code/implementation/docker/storage/src/main/java/io/cattle/platform/docker/storage/DockerImageCredentialLookup.java

```diff
@@ -0,0 +1,49 @@
+package io.cattle.platform.docker.storage;
+
+import io.cattle.platform.core.model.Credential;
+import io.cattle.platform.core.model.StoragePool;
+import io.cattle.platform.docker.client.DockerImage;
+import io.cattle.platform.storage.ImageCredentialLookup;
+import io.cattle.platform.util.type.CollectionUtils;
+
+import java.util.List;
+
+public class DockerImageCredentialLookup implements ImageCredentialLookup {
+    @Override
+    public Credential getDefaultCredential(String uuid, List<?> storagePools, List<?> credentials) {
+        String serverAddress = DockerImage.parse(uuid).getServer();
+        Long registryId = null;
+        for(Object item: storagePools){
+            if (!(item instanceof StoragePool)) {
+                continue;
+            }
+            StoragePool registry = ((StoragePool) item);
+            if (!(registry.getKind().equalsIgnoreCase("registry"))){
+                continue;
+            }
+            if (serverAddress.equalsIgnoreCase((String) CollectionUtils.getNestedValue(registry.getData(), "fields", "serverAddress"))) {
+                registryId = registry.getId();
+                break;
+            }
+        }
+        Credential toReturn = null;
+        for(Object cred: credentials){
+            if (!(cred instanceof Credential)){
+                continue;
+            }
+            Credential credential = (Credential) cred;
+            if (!credential.getKind().equalsIgnoreCase("registryCredential")){
+                continue;
+            }
+            if (credential.getRegistryId().equals(registryId)){
```

Comment:
is there a chance registryId can be null at this point (when the code never hit line 25)? If so, we should probably handle it.